speechbrain/lobes/models/huggingface_transformers/speecht5.py

```diff
@@ -0,0 +1,460 @@
+        if self.output_all_hiddens:
+            encoder_output = encoder_output[-1]
+
+        if self.encoder_only:
+            return encoder_output
+
+        # In the same fashion as mBART:
+        # We replace 0 elements by pax_idx as pad_idx of SpeechT5 is 1
+        decoder_input_ids = custom_padding(
+            x=decoder_input_ids,
+            org_pad=0,
+            custom_pad=self.model.speecht5.decoder.config.pad_token_id,
+        )
+
+        if self.freeze:
+            with torch.no_grad():
+                logits, attentions = self.forward_decoder(
+                    audio_features=encoder_output,
+                    decoder_input_ids=decoder_input_ids,
+                )
+        else:
+            logits, attentions = self.forward_decoder(
+                audio_features=encoder_output,
+                decoder_input_ids=decoder_input_ids,
+            )
+
+        return encoder_output, logits, attentions
+
+    def forward_encoder(self, wav) -> torch.Tensor:
+        """This method implements the forward step of the SpeechT5 encoder for
+        speech recognition
+
+        Arguments
+        ----------
+        wav : tensor or numpy.ndarray
+            wavform of the input.
+
+        Returns
+        -------
+        torch.Tensor
+            Encoder output
+        """
+        if self.output_all_hiddens:
+            if self.freeze_encoder:
+                with torch.no_grad():
+                    states = self.model.speecht5.encoder(
+                        input_values=wav,
+                        output_hidden_states=self.output_all_hiddens,
+                    )
+            else:
+                states = self.model.speecht5.encoder(
+                    input_values=wav,
+                    output_hidden_states=self.output_all_hiddens,
+                )
+            return torch.stack(states.hidden_states)
+
+        elif not self.output_all_hiddens:
+            if self.freeze_encoder:
+                with torch.no_grad():
+                    states = self.model.speecht5.encoder(
+                        input_values=wav,
+                        output_hidden_states=self.output_all_hiddens,
+                    )
+            else:
+                states = self.model.speecht5.encoder(
+                    input_values=wav,
+                    output_hidden_states=self.output_all_hiddens,
+                )
+            return states.last_hidden_state
+
+    def forward_decoder(self, audio_features, decoder_input_ids):
+        """Perform one step of the SpeechT5 decoder.
+
+        Arguments
+        ---------
+        audio_features : torch.Tensor
+            A batch of audio features (SpeechT5 encoding).
+        decoder_input_ids : torch.Tensor
+            A batch of decoder inputs tokens.
+
+        For more details or go to theseq2seq2.py file in SpeechBrain to see how to generate
+        the tokens with Greedy Search and/or Beam Search.
+        """
```

Comment:
im wondering but is there kv cache support on this model with HF ?

Reply:
I do not know, sorry